core/src/main/java/hudson/model/UsageStatistics.java

```diff
@@ -126,51 +125,38 @@ private RSAPublicKey getKey() {
      * Gets the encrypted usage stat data to be sent to the Hudson server.
      */
     public String getStatData() throws IOException {
-        Jenkins j = Jenkins.getInstance();
 
         JSONObject o = new JSONObject();
-        o.put("stat",1);
-        o.put("install", j.getLegacyInstanceId());
-        o.put("servletContainer", j.servletContext.getServerInfo());
-        o.put("version", Jenkins.VERSION);
-
-        List<JSONObject> nodes = new ArrayList<JSONObject>();
-        for( Computer c : j.getComputers() ) {
-            JSONObject  n = new JSONObject();
-            if(c.getNode()==j) {
-                n.put("master",true);
-                n.put("jvm-vendor", System.getProperty("java.vm.vendor"));
-                n.put("jvm-name", System.getProperty("java.vm.name"));
-                n.put("jvm-version", System.getProperty("java.version"));
+        for (UsageStatisticsContributor contributor : UsageStatisticsContributor.all()) {
+            Class<? extends UsageStatisticsContributor> contributorClass = contributor.getClass();
+            LOGGER.finest(String.format("Calling '%s' usage stats contributor",  contributorClass));
+
+            JSONObject contributed = null;
+            try { // robustness: a sole contributor shouldn't be able to interrupt the whole stats retrieval process
+                contributed = contributor.getStatData();
+            } catch (Exception e) {
+                LOGGER.warning(String.format("'%s' threw a '%s 'exception: '%s'.", contributorClass, e, e.getMessage()));
```

Comment:
Better to say

```java
} catch (RuntimeException e) {
    LOGGER.log(Level.WARNING, null, e);
}
```

The information about the class is almost surely redundant given a stack trace.